src/cascadia/TerminalSettingsModel/ElevatedState.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ElevatedState.h"
+#include "CascadiaSettings.h"
+#include "ElevatedState.g.cpp"
+
+#include "JsonUtils.h"
+#include "FileUtils.h"
+
+#include <aclapi.h>
+
+constexpr std::wstring_view stateFileName{ L"elevated-state.json" };
+
+using namespace ::Microsoft::Terminal::Settings::Model;
+
+namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+{
+    ElevatedState::ElevatedState(std::filesystem::path path) noexcept :
+        BaseApplicationState{ path } {}
+
+    // Returns the application-global ElevatedState object.
+    Microsoft::Terminal::Settings::Model::ElevatedState ElevatedState::SharedInstance()

```

Comment:
Perhaps I'm missing something, but wouldn't it be simpler to just have a `ApplicationState::SharedElevatedState()` that uses the separate path and `isElevated` flag, but otherwise shares fields with plain ApplicationState? I don't know how many admin-specific features there will be, but I'm sure people would want to have, e.g. window layout saved in the admin file.

Reply:
omfg that's so much easier it hurts.

Reply:
Plus that would help solve the "persist window state when elevated" problem. The two integrity levels are already basically separate instances entirely, so it makes sense that they would have different app state.